### ISSUE TITLE
Support nvidia T4 ( g4 instances on AWS )

### ISSUE THREAD

**Issue by rguilmont** [jolibrain/deepdetect#675]
Hi, i would love to run DeepDetect on the new G4 AWS instances : https://aws.amazon.com/ec2/instance-types/g4/ . They are using Nvidia T4 for this instances.

Is it possible to support this type of GPU ? Looks like it would require CUDA 10.

### FYI

I managed to compile DeepDetect with CUDA 10 for this series of GPU this way : 

```
    cmake deepdetect \
        -DUSE_CUDNN=ON \
        -DCUDA_ARCH="-gencode arch=compute_75,code=sm_75"
```
Which caused a failure on Caffe build , with : `error: identifier "__shfl_down" is undefined` ( sorry, i don't have the exact log anymore but it's easy to reproduce ).
I "managed" to "fix" it with these changes : https://github.com/rguilmont/caffe/commit/acbdd244078541cf6978fb6884fa671632001ae2

So in the end, deepdetect ran on a g4 machine, and i managed to load models and do prediction without any errors...

... But it looks like there's something weird with GPU memory usage, it takes roughly 3 times more ram for the exact same models loaded on g3s.xlarge instances, or machines with 1080ti. But then we are far from my area of expertise and i have no idea why.

**Comment by fantes** [comment]
thank you  a lot for the report, 
we just PR'ed a change (https://github.com/jolibrain/caffe/pull/59) that should make everything compile finely.  As soon as this PR is merged, you'll have to clean your caffe build within dede and compile dede again with the option you used (it should download the latest caffe archive).
Unfortunately , I cannot reproduce the memory over consumption, I do not have access to  such hardware ...

**Comment by beniz** [comment]
Also @rguilmont you should not use your modified version, as it is passing an empty parameter instead of the mask, as fixed by @fantes .

**Comment by rguilmont** [comment]
Ah yeah, understood ! I'll try as soon as it's merged !

Thanks for quick answer !

**Comment by rguilmont** [comment]
So, i tested your version of caffe, it works great, but i still had the issue of ram overconsumption, but it was completely fixed by switching CUDA 10.2 to 10.1 !

Thanks a lot for your help guys ! Now waiting for caffe to be merged to close the issue.
